### DISCUSSION TITLE
Multiple Connection Writing

### DISCUSSION THREAD

**Discussion by SMDIndomitable** [duckdb/duckdb#10264] [Q&A]
Hi, I am using QThreads from Pyqt5 to open a database connection. I have 4 database files in two different folders (2 db files per folder). I created a thread for each of the folder, and proceed to connect to the 2 db files for each folder. When I am writing to folder 1, the data got written to folder 2 instead. From what I tested, it seems that there can only be 1 existing connection for writing, is this a limitation? 

If so is there any solution for opening multiple connection, 1 connection for each folder, so I am saving to the right database

**Comment by Tishj** [comment]
Some helpful reading:
<https://duckdb.org/faq#how-does-duckdb-handle-concurrency>

<https://duckdb.org/docs/sql/statements/attach>
Using ATTACH you can attach a database with a different mode, allowing you to connect to one database in read/write and to another with only reading capabilities

**Reply by SMDIndomitable** [reply]
Hi, thank you for the reply. Unfortunately, that's not what I am trying to achieve. I am trying to open two connections and the connections are different databases. However, I realise I can only get one connection because all of the data are being saved to the latest connection instead of being separated
